### PR TITLE
feat: authenticate to aspen api using bearer token

### PR DESCRIPTION
**Pull request by asmfstatoil**
Aspen data is now available outside of VPN.

### CODE REVIEW

File: tagreader/web_handlers.py [hunk 1]
```diff
@@ -38,12 +41,26 @@ def get_url_pi():
...
+        return auth
 
 
 def get_url_aspen():
-    # internal url (redirects to dll)
-    return r"https://aspenone.api.equinor.com"
+    if internal:
+        # internal url (redirects to url including AtProcessDataREST.dll)
+        return r"https://aspenone.api.equinor.com"
+    else:
```
Comment (einarsi) [inline tagreader/web_handlers.py:62]: Suggest to remove the else statement - the code below will be executed anyway.

**Review by einarsi — CHANGES_REQUESTED** [review]
Fantastic effort! Please see comments/suggestions.

File: tagreader/web_handlers.py [hunk 2]
```diff
@@ -38,12 +41,26 @@ def get_url_pi():
 
 
 def get_auth_aspen():
-    return HTTPKerberosAuth(mutual_authentication=OPTIONAL)
+    if internal:
+        return HTTPKerberosAuth(mutual_authentication=OPTIONAL)
+    else:
```
[the remark this answers is not in the document]
Comment (asmfstatoil) [reply tagreader/web_handlers.py:44]: makes it an optional argument, it is worthwhile keeping the option of connecting to the internal url and using kerberosauth if proxy is down or there are other issues.

File: tagreader/web_handlers.py (hunk 1 above)
Comment (asmfstatoil) [reply tagreader/web_handlers.py:60]: See above

File: tagreader/web_handlers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (einarsi) [reply tagreader/web_handlers.py:44]: What I meant is that 
```python
if something:
    return whatever
else:
    do_something_else()
```
is functionally equivalent to
```python
if something:
    return whatever
do_something_else()
```
The latter has less indentation and is cleaner. Definitely not a biggie, though.

[the remark this answers is not in the document]
Comment (asmfstatoil) [reply tagreader/web_handlers.py]: done

**Comment by einarsi** [comment]
I was looking into merging this today, but there are a number of tests failing. 13 tests have error messages like this:
```
./tests/test_AspenHandlerREST_connect.py::test_list_sources_aspenone Failed: [undefined]requests.exceptions.SSLError: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /3aa4a235-b6e2-48d5-9195-7fcf05b459b0/v2.0/.well-known/openid-configuration (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)')))
self = <urllib3.connectionpool.HTTPSConnectionPool object at 0x000002652292C730>
method = 'GET'
url = '/3aa4a235-b6e2-48d5-9195-7fcf05b459b0/v2.0/.well-known/openid-configuration'
body = None
headers = {'User-Agent': 'python-requests/2.27.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'}
retries = Retry(total=0, connect=None, read=None, redirect=None, status=None)
redirect = False, assert_same_host = False
timeout = Timeout(connect=None, read=None, total=None), pool_timeout = None
release_conn = False, chunked = False, body_pos = None
response_kw = {'decode_content': False, 'preload_content': False}
parsed_url = Url(scheme=None, auth=None, host=None, port=None, path='/3aa4a235-b6e2-48d5-9195-7fcf05b459b0/v2.0/.well-known/openid-configuration', query=None, fragment=None)
destination_scheme = None, conn = None, release_this_conn = True
http_tunnel_required = True, err = None, clean_exit = False
```
As a sidenote I need to amend the workflows so that we can get proper automated testing of PRs.

**Review by mortendaehli — APPROVED** [review]

**Pull request merged**